### ISSUE TITLE
Prometheus metrics: gauge for current failed syncs

### ISSUE THREAD

**Issue by candlerb** [zrepl/zrepl#338]
zrepl exposes counters for logs at different levels:

```
zrepl_daemon_log_entries{level="error",zrepl_job="backup_storage1"} 9
```

Whilst this is reasonably useful, I think it would be an improvement to have *gauges* for each job showing the number of filesystems which are currently out of sync, i.e. where the most recent replication attempt was a failure.

A dashboard could then show with confidence "this job is / is not in sync", rather than "an error occurred 3 hours ago, but we don't know whether it's still a problem or has been resolved"

I couldn't find an existing metric which shows this.

**Comment by problame** [comment]
Thanks for this suggestion, I like the idea very much and would welcome pull requests.
If anyone is working on this, please let me know in a comment!

---

### Impl Advice

* package replicationg generates a report
* the report is what is used to render the `zrepl status` ui
* Use the `zrepl status --raw` to get an idea of how the data looks like
* Option A (preferred): implement a Getter on `github.com/zrepl/zrepl/replication/report.Report` ( https://github.com/zrepl/zrepl/blob/master/replication/report/replication_report.go#L8 ) that returns the gauge value. I can do the rest of the plumbing to expose the prometheus metrics.
* Option B: Look at the  `zrepl status --raw` JSON output and provide code in `jq` / `python` / whatever that computes the gauge value out of it. I'll translate it to Go code then. (More time consuming for me, less likely to happen / more overhead)

**Comment by problame** [comment]
implemented in #341 , available in zrepl 0.3 (released)
